### PR TITLE
Make the Site header accessible

### LINKED ISSUE #12179: My Site: VoiceOver: site icon is not accessible
Steps to reproduce:

1. Go to Settings > General > Accessibility > VoiceOver and toggle the setting on.
1. Go to My Sties and select a site.
1. Try to double tap the site icon at the top next to the site title and tagline.

**Result:** it's not possible to access the site icon with VoiceOver enabled. ([37s](https://cloudup.com/cW_7txdP1qw))

site-icon-is-not-accessible
<sup>Tested with WPiOS 12.8.1 on iPhone 6S iOS 12.3.1.</sup>

<sub>(internal reference: p77Llu-bQ6-p2 h/t @etoledom)</sub>

### LINKED ISSUE #12180: My Site: VoiceOver: the header on the blog main screen should be one accessible element
Steps to reproduce:

1. Go to Settings > General > Accessibility > VoiceOver and toggle the setting on.
1. Go to My Sties and select a site.
1. Interact with the header.

**Result:** the whole header should be one accessible element.

IMG_2914
<sup>Tested with WPiOS 12.8.1 on iPhone 6S iOS 12.3.1.</sup>

<sub>(internal reference: p77Llu-bQ6-p2 h/t @etoledom)</sub>

### PR DESCRIPTION
**Pull request by shiki**
Closes #12179, and closes #12180. 

This applies accessibility labels to the Site Page's icon, title, and domain. The title and domain are now considered as one and will be triggered if hovering over them and the icon. 

| Main | Site Icon |
|--------|-------|
|  |      |

## Testing

1. Navigate to a Site Page.
2. Turn on VoiceOver. 
3. Hover over the title, domain, and icon. The icon should be dictated as a button. 

This is my first time adding VoiceOver support so I'd be happy to hear your comments. 🙂 

## Reviewing 

Only 1 reviewer is needed but anyone can review. 

## Release Notes

- [x] If there are user-facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/BlogDetailHeaderView.m [hunk 1]
```diff
@@ -63,6 +66,12 @@ - (void)setBlog:(Blog *)blog
         UIDropInteraction *dropInteraction = [[UIDropInteraction alloc] initWithDelegate:self];
         [self.blavatarDropTarget addInteraction:dropInteraction];
     }
+    
+    NSString *localizedLabel =
+        NSLocalizedString(@"Site named %@ located at %@",
```
Comment (etoledom) [inline WordPress/Classes/ViewRelated/Blog/BlogDetailHeaderView.m]: This label sounds a bit too difficult to understand.
What about we simplify it to something like: `@"%@, at %@"`?
Also with the `,` so the screen reader take a breath :)

Comment (shiki) [reply WordPress/Classes/ViewRelated/Blog/BlogDetailHeaderView.m]: Great suggestion! Changed in 2b02be30b52bc85739a5f0d6819ac500c4f6088a.

Also, VoiceOver does not seem to read the domain very well. Do you know if there's any way to improve that?

**Comment by shiki** [comment]
Thank you, @etoledom! Ready for another look. 🙂

**Comment by shiki** [comment]
Thanks, @etoledom!

**Pull request merged**